Automattic/wp-calypso
# Site profiler: Change button CTA text when is in progress

Fixes https://github.com/Automattic/wp-calypso/issues/82240

## Proposed Changes

* Add a `useDelayedQueryTime` hook that generates a `true` value after X milliseconds (default 3000)
* Change the "Check site" button accordling

## Testing Instructions



* On http://calypso.localhost:3000/site-profiler, load a "quick" site. The text should not change. You can have a quick response if you search the same domain two times in a row (it is put in a cache after the first time)
* Try to load a "slow" site (> 3 seconds). The text, should change to "Still checking…" after three seconds
* Try the same site again. The initial text should be again "Check site"

You can simulate a slower site by changing the default limit argument to a lower value: https://github.com/Automattic/wp-calypso/pull/82286/files#diff-c178d0ce1474fd0eb0e487b739841e6b0abf813960c93f1c7b4de394d344223bR4

<img width="878" alt="image" src="https://github.com/Automattic/wp-calypso/assets/167611/521d9f96-4bb5-4ab8-9db5-aca735c0b449">


## Pre-merge Checklist
